### ISSUE TITLE
MaxMSP won't open while Jack Server is active. 

### ISSUE THREAD

**Issue by marijncin** [jackaudio/jack2#719]
Hi, I have a very specific problem. 

I am using Jack audio connection kit in combination with MaxMSP
Platform: Windows 10: 

**My setup**
I have ad_asio JackRouter selected as the driver in MaxMSP. MaxMSP remembers it's last used driver. Meaning that If I start Jack first, it will default to ad_asio JackRouter. 

**Either 1 of these things happens:** 
1- Max never starts
3- Max starts right after I close down the Jack server (like it somehow blocks the app from starting)
3- Max starts up and crashes within a second. 

---------------------------------------------------------------------
**Furtermore:** 
- Windows visual c++ dist kit installed
- It does not matter what is in the maxpatch, even an empty patcher will not load.

What would cause an app to not launch, or crash? 

Jack version 1.9.17 (latest), including JackRouter optional installer. I also tested it on an older version (1.9.11), and it has the same results. 

Steps to reproduce: 
1: Install Jack
2: Install MaxMSP
3: Open Maxmsp and set to jackrouter
4: reopen maxmsp 
5: it may or may not load

**Comment by falkTX** [comment]
Then don't use JACK-Router, it is optional for a reason.
There were a few reports like this before, but uninstalling JACK-Router was more difficult before.
If you uninstall JACK and install again without JACK-Router, does Max work again?

**Comment by vichug** [comment]
yo, i don't know a lot about window 10 or even Jack, but for the max part : you may try to thrash your max preferences (it's easy and mostly harmless, see htere https://www.youtube.com/watch?v=t5CVB6gRMxI ) it has a very small chance to solve your problem

**Comment by marijncin** [comment]
Hi, thanks for your reaction. I do wish to use JackRouter. in MaxMSP there is know way that I know off to create a native jack connection. please let me know if I'm wrong.

**Comment by marijncin** [comment]
> Then don't use JACK-Router, it is optional for a reason.
> There were a few reports like this before, but uninstalling JACK-Router was more difficult before.
> If you uninstall JACK and install again without JACK-Router, does Max work again?

Yes, max will work without Jack-Router off. However, there is no way to make native jack connections in max. Not that I know off. That would be ideal.

**Comment by marijncin** [comment]
> yo, i don't know a lot about window 10 or even Jack, but for the max part : you may try to thrash your max preferences (it's easy and mostly harmless, see htere https://www.youtube.com/watch?v=t5CVB6gRMxI ) it has a very small chance to solve your problem

Thank you, but this does not really work. Even if it worked it would be a pain to have to delete it every time.

**Comment by falkTX** [comment]
If it does not support JACK natively and hangs with JACK-Router, the only other possible way would be a virtual soundcard I think. But I am not quite sure if that is really possible, just in theory.

I would imagine something like... MaxMSP -> virtual soundcard -> some jack client that can read from soundcards -> JACK.

Is there an audio plugin that works on windows and can read from a soundcard and give it to the host?
Carla Plugin Host working in multi-client mode could load such plugin to do that.

This would basically be doing the same as JACK-Router does, just with a few more steps.

**Comment by marijncin** [comment]
> If it does not support JACK natively and hangs with JACK-Router, the only other possible way would be a virtual soundcard I think. But I am not quite sure if that is really possible, just in theory.
> 
> I would imagine something like... MaxMSP -> virtual soundcard -> some jack client that can read from soundcards -> JACK.
> 
> Is there an audio plugin that works on windows and can read from a soundcard and give it to the host?
> Carla Plugin Host working in multi-client mode could load such plugin to do that.
> 
> This would basically be doing the same as JACK-Router does, just with a few more steps.

Thank you for the tip. I will look into it. The strange part is this: I tested it on multiple computers. on some of them, max starts as expected. Others it does not. Could hardware have an effect on the behavior of JackRouter? or conflicting drivers maybe?

**Comment by marijncin** [comment]
> Then don't use JACK-Router, it is optional for a reason.
> There were a few reports like this before, but uninstalling JACK-Router was more difficult before.
> If you uninstall JACK and install again without JACK-Router, does Max work again?

**I recently discovered that it will only crash with an external soundcard.**
- It works with the build-in soundcard using ASIO4ALL
- It works with some external cards like Scarlett 2i4, and Behringer UMC

**This is the card I would like to use:**
 https://www.digigram.com/production-and-sound-cards/lx-ip/ 
MaxMSP crashes when:
- I Select ad_asio JackRouter in MaxMSP, and, Asio4all has this card selected

- This Digigram card has asio support
- this card works with asio4all, and dedicated drivers. I have tested it without jack. 

**When I try this card with the dedicated asio driver,** 
Jack gives me this: "Card already in use" . The is no other software running on my PC. I also de-installed ASIO4ALL to see if it would come available, but the error persists. 

How can the hardware provoke a crash like that?

**Comment by marijncin** [comment]
> Then don't use JACK-Router, it is optional for a reason.
> There were a few reports like this before, but uninstalling JACK-Router was more difficult before.
> If you uninstall JACK and install again without JACK-Router, does Max work again?

**I recently discovered that it will only crash with an external soundcard.**
- It works with the build-in soundcard using ASIO4ALL
- It works with some external cards like Scarlett 2i4, and Behringer UMC

**This is the card I would like to use:**
 https://www.digigram.com/production-and-sound-cards/lx-ip/ 
MaxMSP crashes when:
- I Select ad_asio JackRouter in MaxMSP, and, Asio4all has this card selected

- This Digigram card has asio support
- this card works with asio4all, and dedicated drivers. I have tested it without jack. 

**When I try this card with the dedicated asio driver,** 
Jack gives me this: "Card already in use" . The is no other software running on my PC. I also de-installed ASIO4ALL to see if it would come available, but the error persists. 

How can the hardware provoke a crash like that?

> Then don't use JACK-Router, it is optional for a reason.
> There were a few reports like this before, but uninstalling JACK-Router was more difficult before.
> If you uninstall JACK and install again without JACK-Router, does Max work again?

> Then don't use JACK-Router, it is optional for a reason.
> There were a few reports like this before, but uninstalling JACK-Router was more difficult before.
> If you uninstall JACK and install again without JACK-Router, does Max work again?

**I recently discovered that it will only crash with an external soundcard.**
- It works with the build-in soundcard using ASIO4ALL
- It works with some external cards like Scarlett 2i4, and Behringer UMC

**This is the card I would like to use:**
 https://www.digigram.com/production-and-sound-cards/lx-ip/ 
MaxMSP crashes when:
- I Select ad_asio JackRouter in MaxMSP, and, Asio4all has this card selected

- This Digigram card has asio support
- this card works with asio4all, and dedicated drivers. I have tested it without jack. 

**When I try this card with the dedicated asio driver,** 
Jack gives me this: "Card already in use" . The is no other software running on my PC. I also de-installed ASIO4ALL to see if it would come available, but the error persists. 

How can the hardware provoke a crash like that?

**Comment by marijncin** [comment]
> Then don't use JACK-Router, it is optional for a reason.
> There were a few reports like this before, but uninstalling JACK-Router was more difficult before.
> If you uninstall JACK and install again without JACK-Router, does Max work again?

> Then don't use JACK-Router, it is optional for a reason.
> There were a few reports like this before, but uninstalling JACK-Router was more difficult before.
> If you uninstall JACK and install again without JACK-Router, does Max work again?

**I recently discovered that it will only crash with an external soundcard.**
- It works with the build-in soundcard using ASIO4ALL
- It works with some external cards like Scarlett 2i4, and Behringer UMC

**This is the card I would like to use:**
 https://www.digigram.com/production-and-sound-cards/lx-ip/ 
MaxMSP crashes when:
- I Select ad_asio JackRouter in MaxMSP, and, Asio4all has this card selected

- This Digigram card has asio support
- this card works with asio4all, and dedicated drivers. I have tested it without jack. 

**When I try this card with the dedicated asio driver,** 
Jack gives me this: "Card already in use" . The is no other software running on my PC. I also de-installed ASIO4ALL to see if it would come available, but the error persists. 

How can the hardware provoke a crash like that?

**Comment by 000masa000** [comment]
Try ad_Jackrouter in Audio Status, that seems to work better.
